worldedit-core/src/main/java/com/sk89q/worldedit/extension/factory/parser/pattern/SimplexPatternParser.java

```diff
@@ -0,0 +1,73 @@
+package com.sk89q.worldedit.extension.factory.parser.pattern;
+
+import com.boydti.fawe.object.random.SimplexRandom;
+import com.sk89q.worldedit.WorldEdit;
+import com.sk89q.worldedit.extension.factory.parser.RichParser;
+import com.sk89q.worldedit.extension.input.InputParseException;
+import com.sk89q.worldedit.extension.input.ParserContext;
+import com.sk89q.worldedit.function.pattern.Pattern;
+import com.sk89q.worldedit.function.pattern.RandomPattern;
+
+import java.util.stream.Stream;
+
+public class SimplexPatternParser extends RichParser<Pattern> {
+    private static final String SIMPLEX_PREFIX = "#simplex";
+
+    public SimplexPatternParser(WorldEdit worldEdit) {
+        super(worldEdit, SIMPLEX_PREFIX);
+    }
+
+    @Override
+    protected Stream<String> getSuggestions(String argument, int index) {
+        if (index == 0) {
+            if (argument.isEmpty()) {
+                return Stream.of("1", "2", "3", "4", "5", "6", "7", "8", "9");
+            }
+            // if already a valid number, suggest more digits
+            if (isDouble(argument)) {
+                Stream<String> numbers = Stream.of("", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9");
+                if (argument.indexOf('.') == -1) {
+                    numbers = Stream.concat(numbers, Stream.of("."));
+                }
+                return numbers;
+            }
+            // no valid input anymore
+            return Stream.empty();
+        }
+        if (index == 1) {
+            return worldEdit.getPatternFactory().getSuggestions(argument).stream();
+        }
+        return Stream.empty();
+    }
+
+    @Override
+    protected Pattern parseFromInput(String[] arguments, ParserContext context) {
+        if (arguments.length != 2) {
+            // TODO
```

Comment:
I'm not familiar with the way WE handles stuff like syntax error messages, but I guess there's a better way than that

Reply:
If the length of the input doesn't match the length of the format required then an `InputParseException` should be thrown.
An example would be the following code:
`throw new InputParseException("Clipboard offset needs x,y,z coordinates.");`